### PR TITLE
Increase default dynamic filter collection limits

### PR DESCRIPTION
**Pull request by raunaqmorarka**
## Description

Increase default dynamic filter collection limits

## Additional context and related issues

Potentially alternative to https://github.com/trinodb/trino/pull/16875 

## Release notes

(x) This is not user-visible or docs only and no release notes are required.
( ) Release notes are required, please propose a release note for me.
( ) Release notes are required, with the following suggested text:

```markdown
# Section
* Fix some things. ({issue}`issuenumber`)
```

### CODE REVIEW

**Review by sopel39 — APPROVED** [review]
lgtm % assuming coordinator is not becoming a bottleneck now. Might be worth running concurrency benchmarks.

**Comment by raunaqmorarka** [comment]
> lgtm % assuming coordinator is not becoming a bottleneck now. Might be worth running concurrency benchmarks.

I've added a couple of small tweaks and made the increases smaller based on benchmarks. PTAL again

File: core/trino-main/src/main/java/io/trino/execution/DynamicFilterConfig.java [hunk 1]
```diff
@@ -41,24 +41,24 @@
...
-    private DataSize smallPartitionedMaxSizePerDriver = DataSize.of(10, KILOBYTE);
-    private int smallPartitionedRangeRowLimitPerDriver = 100;
-    private DataSize smallPartitionedMaxSizePerOperator = DataSize.of(100, KILOBYTE);
-    private DataSize smallMaxSizePerFilter = DataSize.of(1, MEGABYTE);
-
-    private int largeBroadcastMaxDistinctValuesPerDriver = 5_000;
-    private DataSize largeBroadcastMaxSizePerDriver = DataSize.of(500, KILOBYTE);
-    private int largeBroadcastRangeRowLimitPerDriver = 10_000;
+    private int smallBroadcastMaxDistinctValuesPerDriver = 1_000;
+    private DataSize smallBroadcastMaxSizePerDriver = DataSize.of(200, KILOBYTE);
```
Comment (sopel39) [inline core/trino-main/src/main/java/io/trino/execution/DynamicFilterConfig.java:45]: why not 100kb (as multiplying everything by 5)?

File: core/trino-main/src/main/java/io/trino/operator/DynamicFilterSourceOperator.java [hunk 2]
```diff
@@ -418,7 +418,7 @@ private ChannelFilter(
                     blockTypeOperators.getEqualOperator(type),
                     blockTypeOperators.getHashCodeOperator(type),
                     blockBuilder,
-                    Math.min(maxDistinctValues, 8192) * 2,
```
Comment (sopel39) [inline core/trino-main/src/main/java/io/trino/operator/DynamicFilterSourceOperator.java]: unrelated change?

**Review by sopel39 — APPROVED** [review]

File: core/trino-main/src/main/java/io/trino/server/DynamicFilterService.java [hunk 3]
```diff
@@ -341,9 +341,12 @@ public boolean isAwaitable()
             @Override
             public TupleDomain<ColumnHandle> getCurrentPredicate()
             {
-                Set<DynamicFilterId> completedDynamicFilters = dynamicFilters.stream()
```
[the remark this answers is not in the document]
Comment (raunaqmorarka) [reply core/trino-main/src/main/java/io/trino/server/DynamicFilterService.java:344]: Each invocation to `getDynamicFilterSummaries` was creating a new map of completed dynamic filters and then we would lookup filter in that map.
I just simplified that to lookup in existing map directly.

File: core/trino-main/src/main/java/io/trino/execution/DynamicFilterConfig.java (hunk 1 above)
Comment (raunaqmorarka) [reply core/trino-main/src/main/java/io/trino/execution/DynamicFilterConfig.java:45]: will update to 100kb for consistency

File: core/trino-main/src/main/java/io/trino/operator/DynamicFilterSourceOperator.java (hunk 2 above)
Comment (raunaqmorarka) [reply core/trino-main/src/main/java/io/trino/operator/DynamicFilterSourceOperator.java]: Once we switch to higher default, the sizing factor on maxDistinctValues doesn't need to be as aggressive.
The previous value of 8192 also turned out to be good in microbenchmarking but a bit over eager in using memory for actual queries. Not big enough to notice any change in peak memory usage but I decided to make it more conservative anyway.

**Comment by mosabua** [comment]
Do we need docs for any of these? Or updates to existing docs?

**Comment by raunaqmorarka** [comment]
> Do we need docs for any of these? Or updates to existing docs?

Some of these properties are mentioned in dynamic filtering docs, we want to avoid advertising these low level knobs as we want users to stick to only using `enable-large-dynamic-filters` if they need to.
